jacquard/experiments/constraints.py

```diff
@@ -216,3 +216,23 @@ def matches_user(self, user, context=None):
             return False
 
         return True
+
+    def is_disjoint_from_constraints(self, other_constraints):
```

Comment:
Currently this is "`True` _if_ disjoint" rather than "`True` _iff_ disjoint". Would enjoy renaming this to clarify that it's best-effort: perhaps `is_provably_disjoint_from_constraints`?

Reply:
(Not sold on the name but something that doesn't imply _iff_ semantics)

Reply:
fe5f20e